### PR TITLE
Abstracted Analytics and added new Events

### PR DESCRIPTION
**Pull request by rafbc**
- Abstracted the Analytics to a different file, 
- added ON_IMPRESSION and ON_VIEW
- layout the event for onError

### CODE REVIEW

File: src/Analytics.js [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+		/**
+		 * Event triggered when the user exists from the fullscreen
+		 * @param {NativeEvent} event
+		 */
+		onExitFullscreen(event) {
+			this.onEvent({'type': PlayerEventTypes.ON_EXIT_FULLSCREEN});
+			this.props.onExitFullscreen && this.props.onExitFullscreen(event);
+		}
+
+		/**
```
Comment (pankaj-pp) [inline src/Analytics.js]: I think the document is not correct for this @raf

File: src/Analytics.js [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+				{...this.props}
+				ref={(player) => this.player = player}
+				onReady={this.onReady.bind(this)}
+				onPlay={this.onPlay.bind(this)}
+				onPause={this.onPause.bind(this)}
+				onEnd={this.onEnd.bind(this)}
+				onProgress={this.onProgress.bind(this)}
+				onEnterFullscreen={this.onEnterFullscreen.bind(this)}
+				onExitFullscreen={this.onExitFullscreen.bind(this)}
+				onError={this.onError.bind(this)}
```
Comment (pankaj-pp) [inline src/Analytics.js]: When is this going to get called?

Comment (rafbc) [reply src/Analytics.js]: @pankaj-pp  when we implement the integration of the onError with the native part. Should come in a follow up request soon.

File: src/Analytics.js (hunk 1 above)
Comment (rafbc) [reply src/Analytics.js]: @pankaj-pp  fixed.

**Pull request merged**